CMakeLists.txt

```diff
@@ -11,6 +11,22 @@ set(OSRM_VERSION_MAJOR 5)
 set(OSRM_VERSION_MINOR 2)
 set(OSRM_VERSION_PATCH 0)
 
+# these two functions build up custom variables:
+#   OSRM_INCLUDE_PATHS and OSRM_DEFINES
+# These variables we want to pass to
+# include_directories and add_definitions for both
+# this build and for sharing externally via pkg-config
+
+function(add_includes includes)
```

Comment:
we should maybe call this function `add_dependency_includes` to emphasize that we use this to expose includes of dependencies.